tests/test_step.py

```diff
@@ -177,3 +200,29 @@ def test_build_config_step_kwarg(mock_step_crds, config_file_step):
     assert config['str1'] == 'from kwarg'
     assert config['str2'] == 'from config'
     assert config['str3'] == 'from crds'
+
+
+def test_logcfg_routing(tmpdir):
+
+    cfg = f"""[*]\nlevel = INFO\nhandler = file:{tmpdir}/myrun.log"""
+
+    logcfg_file = str(tmpdir / 'stpipe-log.cfg')
+
+    with open(logcfg_file,'w') as f:
+        f.write(cfg)
+
+    LoggingPipeline.call(logcfg=logcfg_file)
+
+    logdict = logging.Logger.manager.loggerDict
+    for log in logdict:
+        if not isinstance(logdict[log], logging.PlaceHolder):
+            for handler in logdict[log].handlers:
+                if isinstance(handler, logging.FileHandler):
+                    logdict[log].removeHandler(handler)
+                    handler.close()
+
+    if os.path.isfile(tmpdir / 'myrun.log'):
+        with open(tmpdir / 'myrun.log', 'r') as f:
+            fulltext = '\n'.join([line for line in f])
```

Comment:
Since this a test, we don't want `if` statements.  We want this test to fail if it can't find the log.  Otherwise it will fail silently here, and `fulltext` is not defined outside this `if` block.

```suggestion
    with open(tmpdir / 'myrun.log', 'r') as f:
        fulltext = '\n'.join([line for line in f])
```